### DISCUSSION TITLE
Development on Windows (symlinks issue)

### DISCUSSION THREAD

**Discussion by pratyaksh1610** [organicmaps/organicmaps#3095] [Q&A] [answered]
What I have done so far !

1. Forked and cloned the project on my local system.
2. Set up and changed the versions as mentioned in the documentation.

I was trying to set up project on my local system (windows). I am facing errors in `gradle build sync` . 
I have followed this [Android App Setup](https://github.com/organicmaps/organicmaps/blob/master/docs/INSTALL.md#android-app).

![og1](https://user-images.githubusercontent.com/76740999/182942812-18dbf020-4d8e-4150-969d-c894a91f567f.png)

![og2](https://user-images.githubusercontent.com/76740999/182942957-d7169675-554c-4aa6-b35e-703504972b69.png)

![og3](https://user-images.githubusercontent.com/76740999/182942831-02ce88f9-5a2d-4ec9-9587-50a9458e3d11.png)

**Comment by biodranik** [comment]
You did not follow the instruction and forgot to run configure.sh

**Reply by pratyaksh1610** [reply]
Can you please attach the link of the docs to follow up.
Thanks.

**Reply by biodranik** [reply]
https://github.com/organicmaps/organicmaps/blob/master/docs/INSTALL.md#getting-sources

**Reply by pratyaksh1610** [reply]
I have setup and run the above command and did this all again, but now its showing this error. How to resolve this. I am facing issues.
@biodranik 


![og1](https://user-images.githubusercontent.com/76740999/183116099-53fd826f-3e90-4b2b-9b99-cd71394f65ba.png)


![og2](https://user-images.githubusercontent.com/76740999/183116133-1ebc4836-aeae-45d4-9800-b76a50f2bcbb.png)

![og3](https://user-images.githubusercontent.com/76740999/183116421-7187b9b7-2f9c-4fbb-86de-88f0f482f3ea.png)

**Reply by simple40** [reply]
hey @pratyaksh1610 is your problem solved??
As me too is facing that error..

**Reply by pratyaksh1610** [reply]
@simple40 Not solved yet, I am facing this issue.

**Reply by biodranik** [reply]
Looks like you did not follow instructions and did not enable symlinks in git.

**Reply by pratyaksh1610** [reply]
@biodranik  I did that stuff in the beginning only, still facing this issue.

**Reply by pratyaksh1610** [reply]
@biodranik After this command `cd organicmaps` , I am running `bash ./configure.sh` in git bash only , is this the correct way, and then just following the Android guide and setting up NDK and all that stuff and the emulator at last.
Thanks.

**Reply by pratyaksh1610** [reply]
@biodranik I have also run the scripts , still showing this error.

**Reply by biodranik** [reply]
What is inside this file? `D:\opensourcelorganicmaps\android\res\values-zh-rHK\strings.xml` ?
It should be a symlink:
```
$ ls -la android/res/values-zh-rHK
total 0
drwxr-xr-x   3 user  group    96 Mar 23  2021 .
drwxr-xr-x  89 user  group  2848 Jul 14 08:18 ..
lrwxr-xr-x   1 user  group    28 Mar 20  2016 strings.xml -> ../values-zh-rTW/strings.xml
```

**Reply by pratyaksh1610** [reply]
@biodranik I am getting this way.
```
praty /d/opensource/organicmaps/android (master)
$ ls -la res/values-zh-rHK
total 32
drwxr-xr-x 1 praty 197609  0 Aug  6 16:04 ./
drwxr-xr-x 1 praty 197609  0 Aug  6 16:04 ../
lrwxrwxrwx 1 praty 197609 28 Aug  6 16:04 strings.xml -> ../values-zh-rTW/strings.xml

```

**Reply by pratyaksh1610** [reply]
```
$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean

```

**Reply by pratyaksh1610** [reply]
Now the app is working , it is running on the emulator `API 30`, in case of `API 31` it shows : 
`Error while waiting for device: The emulator process for AVD Pixel_2_API_31 has terminated.`

So I am running on `API 30` emulator.

Frequently it is showing the pop up. 

![ogg](https://user-images.githubusercontent.com/76740999/183248104-41db6bbb-25ea-45c0-910a-5ba3f91975f0.png)

**Reply by biodranik** [reply]
Check what is in logs.

**Reply by simple40** [reply]
why I am getting error in that   **values-zh-rHK\strings.xml file**

**Reply by biodranik** [reply]
Because it should be a symlink. And looks like symlinks are not enabled in your system/git.

**Reply by simple40** [reply]
@biodranik  even after enabling symlink ,it is showing that same error in values-zh-rHK\strings.xml file
you can see here....
![Screenshot 2022-08-18 155645](https://user-images.githubusercontent.com/99835388/185373899-48d946c8-0df4-4000-8dee-1ff6b993d9e9.png)
I have enabled the symlink...

**Reply by biodranik** [reply]
This error is related to the symlink issue. Even if they are enabled in git, they may be disabled system-wide. Please double-check that symlinks work on your system.

**Reply by simple40** [reply]
So can you please tell me how to do that..

**Comment by pratyaksh1610** [comment, marked as answer]
@simple40 
Here's what I did on my system to resolve this issue of symlinks, I will try to make it as simple as possible to help others who may encounter a similar issue step by step.

1. Enabled the developer mode on my laptop in settings( in case of windows 10 `settings -> update and security -> developer mode`).
2. Enable symlinks in git first `git config --global core.symlinks true`
3. Run these commands on git.
`git clone --recurse-submodules https://github.com/organicmaps/organicmaps.git`
`cd organicmaps`
`bash ./configure.sh`
4. On running the last command i.e `bash ./configure.sh`, git will show some paths( probably 3 lines of text, which represent paths of different files) , we need to basically run those files on the desired locations in the file structure of the repo by `double clicking` the file (it will work).
5. Then open project on Android Studio and run the project.


Refer this [INSTALL.md](https://github.com/organicmaps/organicmaps/blob/master/docs/INSTALL.md) (official docs) in case of commands mentioned above to avoid error.

**Reply by simple40** [reply]
okay! I will try with these steps

**Reply by simple40** [reply]
now it is giving me the file not found errors..
![Screenshot (30)](https://user-images.githubusercontent.com/99835388/185785122-ccedf726-d884-412f-8337-de52643a6ae6.png)
![Screenshot (31)](https://user-images.githubusercontent.com/99835388/185785130-ead18a79-7c5d-4606-b563-130ef34e024c.png)
![Screenshot (32)](https://user-images.githubusercontent.com/99835388/185785132-9126de09-2fb1-4f39-a9cf-95f0776d02c9.png)

and also after running bash ./configure.sh
I run the bootstrap.bat but there is no the b2 headers file
![Screenshot (33)](https://user-images.githubusercontent.com/99835388/185785455-9e422bef-77f4-4043-99ff-4e4223c60402.png)

can you tell me what I have done wrong??....

**Reply by pratyaksh1610** [reply]
yes these paths are correct that u are getting in cli

**Reply by pratyaksh1610** [reply]
\organicmaps\3party\boost
here run bootstrap.bat

then run
run b2.exe

and bootstrap.sh
and restart android studio

**Reply by simple40** [reply]
I ran those bootstrap.bat and bootstrap.sh files 
but there is no b2.exe inside \organicmaps\3party\boost 
you can see here..
![Screenshot (35)](https://user-images.githubusercontent.com/99835388/185808916-726ffca4-9afd-4c71-aca7-572d4e7a49be.png)

and also getting this now 

![Screenshot (34)](https://user-images.githubusercontent.com/99835388/185808955-9acf24f1-f311-48b6-b7fa-27119f4395c9.png)

and thank you for helping me **: )**

**Comment by pratyaksh1610** [comment]
I am getting these files, so acc. to this you need to run 
`bootstrap.bat` `bootstrap.sh`  `b2.exe`

![bba](https://user-images.githubusercontent.com/76740999/185847231-911d46fb-b9bc-423a-a7fc-4090ba51e76f.png)

**Reply by simple40** [reply]
@pratyaksh1610 
@biodranik 
getting this on running bootstrap.bat

![Screenshot (36)](https://user-images.githubusercontent.com/99835388/185986287-8fa8ebfc-4f76-4100-a634-ead3c5cd97ad.png)

why it got failed???
and also b2.exe is not present there and I think it is due the above error
and don't know what should I do.

**Reply by pratyaksh1610** [reply]
You need to run these files by just double clicking the file at the desired location.

**Reply by pratyaksh1610** [reply]
https://github.com/organicmaps/organicmaps/discussions/3095#discussioncomment-3436114 Follow this , this might be the correct answer to your problem as I have written steps in detail.

**Reply by pratyaksh1610** [reply]
Just delete repo from local system and follow above steps in order starting from fork and all that stuff, this will surely solve the issue.

**Reply by adityarai004** [reply]
hi
i have followed all the steps and havent missed any of them but still there is an error from last 6 hours 
![image](https://user-images.githubusercontent.com/86659578/219098408-e1c7d809-82ea-4115-8c22-a43462775530.png)
I LITERALLY HAVE DONE EVERYTHING PLEASE DO NOT LINK ME ANYWHERE THAT FOLLOW THESE STEPS AND THAT I HAVE DONE EVERY EVERY THING PLEASE
